roles/docker/tasks/install.yml

```diff
@@ -0,0 +1,123 @@
+---
+
+- name: update apt cache
+  apt:
+    update_cache: true
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: Delete previous apt sources (1)
+  ansible.builtin.file:
+    path: /etc/apt/sources.list.d/docker.list
+    state: absent
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: Delete previous apt sources (2)
+  ansible.builtin.file:
+    path: /etc/apt/sources.list.d/download_docker_com_linux_ubuntu.list
+    state: absent
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: install required system packages
+  apt:
+    pkg:
+      - apt-transport-https
+      - ca-certificates
+      - curl
+      - software-properties-common
+      - python3-pip
+      - virtualenv
+      - python3-setuptools
+    state: latest
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: Create directory for Docker's GPG key
+  file:
+    path: /etc/apt/keyrings
+    state: directory
+    mode: '0755'
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: Download Docker GPG key
+  get_url:
+    url: https://download.docker.com/linux/ubuntu/gpg
+    dest: /etc/apt/keyrings/docker.asc
+    mode: '0644'
+    owner: root
+    group: root
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: Add Docker repository to APT sources
+  vars:
+    arch_mapping:
+      x86_64: amd64
+      aarch64: arm64
+  apt_repository:
+    repo: "deb [arch={{ arch_mapping[ansible_architecture] | default(ansible_architecture) }} signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu {{ ansible_lsb.codename }} stable"
+    state: present
+    filename: docker
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: update apt cache
+  apt:
+    update_cache: true
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: install docker-ce
+  apt:
+    name: docker-ce
+    state: latest
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: install docker module for Python
+  pip:
+    name: docker
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: pull hello-world docker image
+  community.docker.docker_image:
+    name: "hello-world"
+    source: pull
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
+
+- name: create hello-world container
+  community.docker.docker_container:
+    name: "hello-world"
+    image: "hello-world"
+    state: started
+    auto_remove: true
+  when: inventory_hostname in groups['srsran_nodes']
+  become: true
```

Comment:
You can use a task like this one to verify whether `Docker` is already installed. If so, then, do not installed it (again)

Reply:
Thanks. I have used different mechanism to check docker availability and skip the docker install if the system already installed with docker.

Reply:
We may need to think whether checking Docker (docker --help) is enough for it to be installed? or whether actually pulling/running the `hello-world` image is more appropriate. Whatever approach we take, we should use it across all blueprints that needed it